### PR TITLE
OTLP HTTP Metrics Exporter

### PR DESCRIPTION
**Pull request by lalitb**
Issue: #772 

Had a quick and naïve attempt on adding support for OTLP HTTP Metrics Exporter. The initial motive was more to ramp-up on Rust, and understand otel-rust setup/build.

If this seems helpful and in right direction, I can spend some more time next week on adding unit-test, fix lint/warnings, fix CI build issues to make it ready for review.  Or if this doesn't look to be right approach, or someone is already working on enabling otlp-http, please let me know I will just abandon the PR :)

### CODE REVIEW

**Comment by TommyCpp** [comment]
👍 I think it's on the right track. Thanks for working on this

**Comment by lalitb** [comment]
> 👍 I think it's on the right track. Thanks for working on this

Thanks, will make it ready for review shortly.

**Comment by jtescher** [comment]
@lalitb If you merge in the changes from the main branch it should fix your CI issues.

**Comment by lalitb** [comment]
> @lalitb If you merge in the changes from the main branch it should fix your CI issues.

@jtescher yes it did fix after merging from main. Thanks

**Review by hdost — APPROVED** [review]
Thank you @lalitb 
@open-telemetry/rust-approvers  Can we make sure this gets squashed when merged, there's a lot of "change" commits.

**Comment by lalitb** [comment]
> Thank you @lalitb @open-telemetry/rust-approvers Can we make sure this gets squashed when merged, there's a lot of "change" commits.

Thanks @hdost. Was initially not thinking to raise PR while I was working on it, so commit comments were not thought of. Good to squash them while merging.

**Review by TommyCpp — APPROVED** [review]
Can we add a change log? Otherwise LGTM and I can help merging it

File: examples/basic-otlp-http/src/main.rs [hunk 1]
```diff
@@ -19,14 +22,44 @@ fn init_tracer() -> Result<sdktrace::Tracer, TraceError> {
...
+fn init_metrics() -> metrics::Result<sdkmetrics::MeterProvider> {
+    let export_config = opentelemetry_otlp::ExportConfig {
+        endpoint: "http://localhost:4318/v1/metrics".to_string(),
+        ..opentelemetry_otlp::ExportConfig::default()
+    };
+    opentelemetry_otlp::new_pipeline()
+        .metrics(opentelemetry_sdk::runtime::Tokio)
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .http()
```
Comment (cijothomas) [inline examples/basic-otlp-http/src/main.rs:34]: https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/src/lib.rs#L254 You can now remove the comment about lack of metric support.

File: opentelemetry-otlp/src/metric.rs [hunk 2]
```diff
@@ -303,13 +386,83 @@ impl MetricsExporter {
...
+            Ok(val) => val,
+            Err(_) => format!("{}{}", config.endpoint, "/v1/metrics"),
+        };
+
+        let url: Uri = config
+            .endpoint
+            .parse()
+            .map_err::<crate::Error, _>(Into::into)?;
+
+        let _timeout = match std::env::var(OTEL_EXPORTER_OTLP_METRICS_TIMEOUT) {
```
Comment (cijothomas) [inline opentelemetry-otlp/src/metric.rs:416]: not using `_timeout`?

**Review by cijothomas — APPROVED** [review]
left a couple of nits.

File: opentelemetry-otlp/src/metric.rs (hunk 2 above)
Comment (lalitb) [reply opentelemetry-otlp/src/metric.rs:416]: Yes as of now, it is using the timeout from config file (consistent with the otlp-http span exporter). Thought of creating separate PR to read _timeout from environment variable.

File: examples/basic-otlp-http/src/main.rs (hunk 1 above)
Comment (lalitb) [reply examples/basic-otlp-http/src/main.rs:34]: Thanks done.

**Comment by lalitb** [comment]
> Can we add a change log? Otherwise LGTM and I can help merging it

Thanks updated changelog.

**Comment by cijothomas** [comment]
@hdost @TommyCpp Could you help merge this?

**Comment by hdost** [comment]
Sadly i can only Approve at the moment.

**Pull request merged**